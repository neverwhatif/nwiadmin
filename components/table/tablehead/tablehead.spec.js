import React from 'react';
import { shallow } from 'enzyme';

import TableHead from './index';

const data = {
    $id: 1,
    Title: 'Lorem',
};

describe('TableHead', () => {
    const component = shallow(<TableHead data={data} />);

    it('should render', () => {
        expect(component).toMatchSnapshot();
    });
    it('should render cells', () => {
        expect(component.find('th').length).toBe(1); // Don't include $id
        expect(component.find('th').first().text()).toBe('Title');
    });
});
