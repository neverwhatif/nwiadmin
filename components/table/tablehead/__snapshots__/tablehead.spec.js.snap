// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHead should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TableHead
    data={
      Object {
        "$id": 1,
        "Title": "Lorem",
      }
    }
    headClass={null}
    isAllSelected={false}
    toggleSelectAll={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          null,
          <th
            className="cell"
          >
            Title
          </th>,
        ],
        undefined,
      ],
      "className": "root",
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": "Title",
        "nodeType": "host",
        "props": Object {
          "children": "Title",
          "className": "cell",
        },
        "ref": null,
        "rendered": "Title",
        "type": "th",
      },
      undefined,
    ],
    "type": "tr",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            null,
            <th
              className="cell"
            >
              Title
            </th>,
          ],
          undefined,
        ],
        "className": "root",
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": "Title",
          "nodeType": "host",
          "props": Object {
            "children": "Title",
            "className": "cell",
          },
          "ref": null,
          "rendered": "Title",
          "type": "th",
        },
        undefined,
      ],
      "type": "tr",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
